merkletree/proof.go

```diff
@@ -16,6 +23,75 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
+	leafIndex, leafCommitment []byte, leafLevel int, isLeafEmpty bool) (leafHash []byte) {
+	if isLeafEmpty {
```

Comment:
> Maybe it also has some advantages in performance (e.g., smaller stack frame ...) :D . 

Haha, this is true. I wonder if it's worth going back and changing the STR and TB verify functions for this reason. Maybe if we notice the memory footprint of these functions being too big, we should.